lstchain/mc/plot_utils.py

```diff
@@ -0,0 +1,162 @@
+
+def fill_bin_content(ax, energy_bin, gb, tb):
+    """
+
+    Parameters
+    --------
+
+    Returns
+    --------
+    """
+    for i in range(0,gb):
+        for j in range(0,tb):
+            text = ax.text((j+0.5)*0.05, (i+0.5)*0.1, "%.2f %%" % sens[energy_bin][i][j],
+                       ha="center", va="center", color="w")
+    return ax
+
+
+def format_axes(ax,pl):
+    """
+
+    Parameters
+    --------
+
+    Returns
+    --------
+    """
+    ax.set_aspect(0.5)
+
+    ax.set_ylabel(r'Gammaness', fontsize=15)
+    ax.set_xlabel(r'$\theta^2$ (deg$^2$)', fontsize=15)
+    
+    starty, endy = ax.get_ylim()
+    ax.yaxis.set_ticks(np.arange(endy, starty, 0.1)[::-1])
+    startx, endx = ax.get_xlim()
+    ax.xaxis.set_ticks(np.arange(startx, endx, 0.1))
+
+    cbaxes = fig.add_axes([0.9, 0.125, 0.03, 0.755])
+    cbar = fig.colorbar(pl, cax=cbaxes)
+    cbar.set_label('Sensitivity (% Crab)', fontsize=15)
+
+
+def format_axes_array(ax, arr_i,arr_j, plot):
+    """
+
+    Parameters
+    --------
+
+    Returns
+    --------
+    """
+    ax.set_aspect(0.5)
+    if ((arr_i == 0) and (arr_j == 0)):
+        ax.set_ylabel(r'Gammaness', fontsize=15)
+    if ((arr_i == 3) and (arr_j == 2)):
+        ax.set_xlabel(r'$\theta^2$ (deg$^2$)', fontsize=15)
+
+    starty, endy = ax.get_ylim()
+    ax.yaxis.set_ticks(np.arange(endy, starty, 0.1)[::-1])
+    startx, endx = ax.get_xlim()
+    ax.xaxis.set_ticks(np.arange(startx, endx, 0.1))
+    
+    cbaxes = fig.add_axes([0.91, 0.125, 0.03, 0.755])
+    cbar = fig.colorbar(plot, cax=cbaxes)
+    cbar.set_label('Sensitivity (% Crab)', fontsize=15) 
+
+def format_axes_sensitivity(ax):
+    """
+
+    Parameters
+    --------
+
+    Returns
+    --------
+    """
+
+    ax.set_xscale("log", nonposx='clip')
+    ax.set_yscale("log", nonposy='clip')
+    ax.set_xlim(5e1, 9.e4)
+    ax.set_ylim(1.e-14, 5.e-10)
+    ax.set_xlabel("Energy [GeV]")
+    ax.set_ylabel(r'E$^2$ $\frac{\mathrm{dN}}{\mathrm{dE}}$ [TeV cm$^{-2}$ s$^{-1}$]')
+    ax.grid(ls='--', alpha=.5)
+
+def plot_Crab_SED(ax, emin, emax, percentage=100, **kwargs):
+    """
+
+    Parameters
+    --------
+
+    Returns
+    --------
+    """
+    En = np.logspace(np.log10(emin.to_value()), np.log10(emax.to_value()), 40) * u.GeV  
+
+    dFdE = percentage / 100. * crab_HEGRA(En)
+    ax.loglog(En, dFdE * En * En, color='gray', **kwargs)
+    
+    return ax
+
+def plot_sensitivity(ax, e, sensitivity):
+    """
+
+    Parameters
+    --------
+
+    Returns
+    --------
+    """
+    emed = np.sqrt(e[1:] * e[:-1])
+    dFdE = crab_HEGRA(emed)
+    ax.loglog(e, sensitivity / 100 * dFdE * emed * emed, \
+              label = 'Sensitivity')
+    
+
+def sens_minimization_plot(eb, gb, tb, e, sens):
+    """
+    TODO: Save plots!
+    Parameters
+    --------
+
+    Returns
+    --------
+    """
+    if (eb == 12):
+        figarr, axarr = plt.subplots(4,3, sharex=True, sharey=True, figsize=(13.2,18))
+
```

Comment:
figarr is not defined anywhere for eb!=12

Reply:
I defined it outside the if statement, but this should be adapted to the number of bins...